### DISCUSSION TITLE
Difference between matRad_calcParticleDoseMC and matRad_calcDoseDirectMC

### DISCUSSION THREAD

**Discussion by ZengYL99** [e0404/matRad#700] [Q&A] [answered]
Hi,

I ran the **matRad_example12_simpleParticleMonteCarlo.m** in **matRad-dev_varRBErobOpt** using the MCsquare engine. The calculation was fast when utilizing `resultGUI_MC = matRad_calcDoseDirectMC(ct,stf,pln,cst,resultGUI.w)`. However, the calculation significantly slowed down when using `dij = matRad_calcParticleDoseMC(ct, stf, pln, cst)`.
I have checked the `dij = matRad_calcParticleDoseMCsquare(ct,stf,pln,cst,calcDoseDirect)`, and found that the primary difference in setting **calcDoseDirect** to true or false before starting MCsquare is the modification of `stfMCsquare(i).energyLayer(k).MU`. Could this be the reason for the slower computation when using `matRad_calcParticleDoseMC` compared to `matRad_calcDoseDirectMC`?

Best regards,
Zeng.

**Comment by wahln** [comment, marked as answer]
Both functions are doing different things for different purposes.
`matRad_calcDoseDirectMC` does a _forward_ dose calculation, i.e., takes a plan with optimized weights and runs a Monte Carlo simulation. It will directly return a dose cube within matRad result structure `resultGUI`. As the beamlet weights are known, significantly less histories are needed (as some beamlets will have very low weights) than in the second case:
`matRad_calcParticleDoseMC` will compute the _dose influence matrix_, returned as the `dij` struct, which can be used for optimization to determine the weights. As the weights are not known yet, each beamlet of the plan needs to be calculated to sufficient accuracy, requiring significantly more MC histories than the forward calculation.

**Reply by wahln** [reply]
One more comment: The `calcDoseDirect` functions as an internal switch to change the behavior of the dose accumulation (either in a dose cube, or in a dose influence matrix).

**Reply by ZengYL99** [reply]
Thank you for your reply. It is very helpful for me! 
Additionally, I have another question regarding MCsquare. It seems that matRad supports parallel calculations for Monte Carlo simulation as it configured the following parameters:
```
pln.propMC.parallelRuns = true;
pln.propMC.numThreads = 8;
```
However, the parallel computing remains inactive after configuring these parameters. Do you have any insights or suggestions on this issue? Your assistance in this matter would be highly appreciated.

**Reply by wahln** [reply]
parallelRuns has no meaning for our MCsquare interface, it is only considered by TOPAS.
MCsquare is tuned for multi-core CPUs and handles highly-optimized multicore parallelization internally. If your CPU still is idling on some cores, increase the number of threads.
if you want to run separate simulations on multiple nodes / CPUs, this is not really supported within our MCsquare interface.

**Reply by ZengYL99** [reply]
I got it! Thank you very much! :)
